mslice/models/cut/matplotlib_cut_plotter.py

```diff
@@ -19,27 +21,37 @@ def plot_cut(self, selected_workspace, cut_axis, integration_axis, norm_to_one,
                                  integration_axis.end)
         self.plot_cut_from_xye(x, y, e, cut_axis.units, selected_workspace, (intensity_start, intensity_end),
                                plot_over, output_ws_name, legend)
+        plt.show()
 
+    @plt.set_category(plt.CATEGORY_CUT)
     def plot_cut_from_xye(self, x, y, e, x_units, selected_workspace, intensity_range=None, plot_over=False,
                           cut_ws_name=None, legend=None):
         legend = selected_workspace if legend is None else legend
-        plt.errorbar(x, y, yerr=e, label=legend, hold=plot_over, marker='o', picker=picker)
-        plt.ylim(*intensity_range) if intensity_range is not None else plt.autoscale()
-        leg = plt.legend(fontsize='medium')
+        if not plot_over:
+            plt.cla()
+        plt.errorbar(x, y, yerr=e, label=legend, marker='o', picker=PICKER_TOL_PTS)
+        cur_fig = plt.gcf()
+        cur_axes = cur_fig.gca()
+        cur_axes.set_ylim(*intensity_range) if intensity_range is not None else cur_axes.autoscale()
+        leg = cur_axes.legend(fontsize='medium')
         leg.draggable()
-        plt.xlabel(get_display_name(x_units, get_comment(selected_workspace)), picker=picker)
-        plt.ylabel(CUT_INTENSITY_LABEL, picker=picker)
+        cur_axes.set_xlabel(get_display_name(x_units, get_comment(selected_workspace)), picker=PICKER_TOL_PTS)
+        cur_axes.set_ylabel(CUT_INTENSITY_LABEL, picker=PICKER_TOL_PTS)
+        cur_canvas = cur_fig.canvas
         if not plot_over:
-            plt.gcf().canvas.set_window_title('Cut: ' + selected_workspace)
-            plt.gcf().canvas.manager.update_grid()
-        if not plt.gcf().canvas.manager.has_plot_handler():
+            cur_canvas.set_window_title('Cut: ' + selected_workspace)
+            cur_canvas.manager.update_grid()
+        if not cur_canvas.manager.has_plot_handler():
             self._create_cut(cut_ws_name if cut_ws_name is not None else selected_workspace)
-        plt.gcf().canvas.restore_region(plt.gcf().canvas.manager.get_cut_background())
+            cur_canvas.restore_region(cur_canvas.manager.get_cut_background())
         try:
-            plt.gca().draw_artist(plt.gcf().canvas.figure.get_children()[1])
-            plt.gcf().canvas.blit(plt.gcf().canvas.figure.gca().clipbox)
+            children = cur_fig.get_children()
+            for artist in children:
+                cur_axes.draw_artist(artist)
+                cur_canvas.blit(cur_axes.clipbox)
```

Comment:
You can move the `blit` call outside the loop, it only needs to be done once.

Reply:
Good catch. I hate Python's indentation sometimes...